File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        if s1 == s2:
            return True
        
        n = len(s1)
        first = -1
        second = -1
        for i in range(n):
            if s1[i] != s2[i]:
                if first == -1:
                    first = i
                elif second == -1:
                    second = i
                else:
                    return False
        return s1[first] == s2[second] and s1[second] == s2[first]

File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.rs
impl Solution {
    pub fn are_almost_equal(s1: String, s2: String) -> bool {
        let arr: Vec<_> = s1.chars().zip(s2.chars())
            .filter(|(a,b)| a != b)
            .collect();
        
        match arr.len() {
            0 => true,
            2 => arr[0].0 == arr[1].1 && arr[0].1 == arr[1].0,
            _ => false,
        }
    }
}